### ISSUE TITLE
Setup Monitoring with InfluxDB & Grafana doesn't work 

### ISSUE THREAD

**Issue by Lilith1410** [ethereum/py-evm#1944]
* py-evm Version: all of them
* OS: Ubuntu 18.04
* Python Version (python --version): 3.7
* Environment (output of `pip freeze`):
aiohttp==3.6.0
anyio==1.3.0
argcomplete==1.11.1
asks==2.3.7
asn1crypto==1.3.0
async-exit-stack==1.0.1
async-generator==1.10
async-lru==0.1.0
async-service==0.1.0a7
async-timeout==3.0.1
asyncio-cancel-token==0.2.0
asyncio-run-in-process==0.1.0a4
attrdict==2.0.1
attrs==19.3.0
backcall==0.1.0
base58==1.0.3
blake2b-py==0.1.3
bloom-filter==1.3
cached-property==1.5.1
cachetools==3.1.1
certifi==2020.4.5.1
cffi==1.14.0
chardet==3.0.4
cloudpickle==1.4.1
coincurve==10.0.0
contextvars==2.4
cryptography==2.6.1
cytoolz==0.10.1
dataclasses==0.7
decorator==4.4.2
eth-abi==2.1.1
eth-account==0.4.0
eth-bloom==1.0.3
eth-hash==0.2.0
eth-keyfile==0.5.1
eth-keys==0.2.4
eth-rlp==0.1.2
eth-typing==2.2.1
eth-utils==1.9.0
fastecdsa==1.7.4
h11==0.9.0
hexbytes==0.2.0
idna==2.7
idna-ssl==1.1.0
ifaddr==0.1.6
immutables==0.14
importlib-metadata==1.6.0
ipfshttpclient==0.4.13.2
ipython==7.9.0
ipython-genutils==0.2.0
jedi==0.17.0
jsonschema==3.0.1
lahja==0.15.2
libp2p==0.1.4
lru-dict==1.1.6
lxml==4.5.1
multiaddr==0.0.9
multidict==4.7.6
mypy==0.770
mypy-extensions==0.4.3
netaddr==0.7.19
netdisco==2.6.0
netifaces==0.10.9
outcome==1.0.1
parsimonious==0.8.1
parso==0.7.0
pexpect==4.8.0
pickleshare==0.7.5
plyvel==1.1.0
prometheus-client==0.7.1
prompt-toolkit==2.0.10
protobuf==3.12.2
ptyprocess==0.6.0
py-ecc==1.7.1
py-evm==0.3.0a14
pycparser==2.20
pycryptodome==3.9.7
pyethash==0.1.27
Pygments==2.6.1
pymultihash==0.8.2
PyNaCl==1.3.0
pyrsistent==0.15.7
pysha3==1.0.2
python-dateutil==2.8.1
python-snappy==0.5.4
requests==2.23.0
rlp==1.2.0
rpcudp==3.0.0
ruamel.yaml==0.15.98
six==1.15.0
sniffio==1.1.0
sortedcontainers==2.1.0
SQLAlchemy==1.3.17
ssz==0.2.3
termcolor==1.1.0
toolz==0.10.0
traitlets==4.3.3
trie==1.4.0
trinity==0.1.0a35
trio==0.13.0
trio-typing==0.3.0
typed-ast==1.4.1
typing-extensions==3.7.4.2
u-msgpack-python==2.6.0
uPnPClient==0.0.8
urllib3==1.25.9
uvloop==0.14.0
varint==1.0.2
wcwidth==0.1.9
web3==5.4.0
websockets==8.1
yarl==1.4.2
zeroconf==0.27.0
zipp==3.1.0

### What is wrong?
I had followed the 'Contributing' Guide on how to setup Trinity so that I can work on the GitHub Repo on my local machine. Then I initially setup the monitoring environment of InfluxDB and Grafana locally and wrote the corresponding guide to that for the docs. 
Now I ran into issues with my Trinity Setup while checking out other commits from the repo and form one day to the other I couldn't start trinity anymore and I have no idea what I did to cause that. Ultimately, after retracing all my steps to get back to my initial (working) setup and fiddeling around with it, I decided to delete the whole folder with the repo and clone again, following the guide but now I am struggling to make it work again. 

I used the command 
`trinity --enable-metrics --metrics-host "test" --metrics-influx-server "localhost" --metrics-influx-password "trinity" --metrics-influx-protocol http --metrics-influx-port 8086` 
but the error message is: argument subcommand: invalid choice: 'test' (choose from 'attach', 'db-shell', 'fix-unclean-shutdown', 'remove-network-db', 'export', 'import')

Also under `trinity --help` there is the no mention of the flags `--metrics-host`, `--metrics-influx-server`, `--metrics-influx-password`, `metrics-influx-protocol http` or `--metrics-influx-port 8086` that I used before and wrote down in the guide. 

### How can it be fixed
I don't know :( I've tried to fix it for several days now. Can anyone help me please?

**Comment by cburgdorf** [comment]
>but the error message is: argument subcommand: invalid choice: 'test' (choose from 'attach', 'db-shell', 'fix-unclean-shutdown', 'remove-network-db', 'export', 'import')

The error message seems weird. It would suggest "test" is used as a subcommand as in `trinity test`

>Also under trinity --help there is the no mention of the flags `--metrics-host, --metrics-influx-server`

This seems to be further evidence that something is off because I do see them listed.

```
$ trinity --help
(...omitting irrelevant parts)
metrics:
  --enable-metrics      Enable metrics component
  --metrics-host METRICS_HOST
                        Host name to tag the metrics data (e.g. trinity-
                        bootnode-europe-pt)
  --metrics-influx-user METRICS_INFLUX_USER
                        Influx DB user. Defaults to `trinity`
  --metrics-influx-database METRICS_INFLUX_DATABASE
                        Influx DB name. Defaults to `trinity`
  --metrics-influx-password METRICS_INFLUX_PASSWORD
                        Influx DB password. Defaults to ENV var
                        TRINITY_METRICS_INFLUX_DB_PW
  --metrics-influx-server METRICS_INFLUX_SERVER
                        Influx DB server. Defaults to ENV var
                        TRINITY_METRICS_INFLUX_DB_SERVER
  --metrics-influx-port METRICS_INFLUX_PORT
                        Influx DB port. Defaults to ENV var
                        TRINITY_METRICS_INFLUX_DB_PORT or 8086
  --metrics-influx-protocol METRICS_INFLUX_PROTOCOL
                        Influx DB protocol. Defaults to ENV var
                        TRINITY_METRICS_INFLUX_DB_PROTOCOL or http
  --metrics-reporting-frequency METRICS_REPORTING_FREQUENCY
                        The frequency in seconds at which metrics are reported
  --metrics-system-collector-frequency METRICS_SYSTEM_COLLECTOR_FREQUENCY
                        The frequency in seconds at which system metrics are
                        collected
  --metrics-blockchain-collector-frequency METRICS_BLOCKCHAIN_COLLECTOR_FREQUENCY
                        The frequency in seconds at which blockchain metrics
                        are collected

...
```

That said, I also run in an error albeit a different one.

>trinity.exceptions.MetricsReportingError: No 'trinity' process found.

I have an idea where this is coming from. @njgheorghita recently added a metric to track the number of processes but the error seems to suggest that this doesn't work on my Ubuntu system. I believe Nick has tested this on a Mac and it probably behaves differently there.

Can you do me a favor and post the following two things to help us debug your case:

1. `git rev-parse HEAD` (to figure out the exact commit you are on)

2. The shell output including your exact command and the following swall of errors you are getting.

**Comment by njgheorghita** [comment]
@cburgdorf Yup - you nailed it. I have what I believe is a fix for the `No 'trinity' process found` in https://github.com/ethereum/trinity/pull/1765 - I've tested it on ubuntu instances on aws - but, it would be maybe helpful if you could verify that it works on your machine?

@Lilith1410 Once when I forgot to activate my virtual environment, I got that exact same bug. But, after entering the virtual environment, everything started working normally. I'm not sure if that's the issue with your machine, but I'd guess it points towards some outdated dependencies/ misconfigured environment.

**Comment by Lilith1410** [comment]
@cburgdorf sure! 
`git rev-parse HEAD` gives me: e40aac0ab2a9a2e7a703af0b94fafd1047ce6e55
The command i am using is `trinity --enable-metrics --trinity-tmp-root-dir --metrics-host "test" --metrics-influx-server "localhost" --metrics-influx-password "trinity"`. 
The error message is `trinity: error: argument subcommand: invalid choice: 'test' (choose from 'attach', 'db-shell', 'fix-unclean-shutdown', 'remove-network-db', 'export', 'import')` 

I have checked Gitkraken again. Currently, I am on the lastest git commit. But I have tried out several in between the last alpha-release and today (two days ago) but so far none I have tried, worked. 

@njgheorghita I tried again this morning :) started influxd and grafana-server services as usual, then `. venv/bin/activate` to activate the virtual environment. then the command listed above to start trinity with the metrics enabled.
